setup.py

```diff
@@ -34,6 +34,7 @@ def get_long_description():
     namespace_packages=[],
     include_package_data=True,
     zip_safe=False,
+    python_requires='>=2.6, !=3.0.*, !=3.1.*, !=3.2.*',
```

Comment:
I would rather this be `>= 2.7` also `!= 3.3.*`

Reply:
I am inclined to leave this as proposed (`python_requires='>=2.6, !=3.0.*, !=3.1.*, !=3.2.*'`), as that matches (in addition to our trove classifiers) what we test for with tox and travis.

@sigmavirus24, what is your rationale for removing 2.6 and 3.3.x ? I am inclined to want to leave them as long as they don't add any additional maintenance overhead or prevent new development...

Reply:
2.6 and 3.3 are no longer supported by the packaging tools we rely on. setuptools, wheel, and other libraries (I think even pip) have dropped support for these EOL'd versions of Python. We will need additional work to actually support them in our CI and other systems. I don't think it's worth the work any longer.

Further, using BigQuery we can find the most popular downloads:

```
❯❯❯ pypinfo -d 90 -pc -md pycodestyle pyversion
```

| python_version | percent | download_count |
| -------------- | ------: | -------------: |
| 2.7            |  44.77% |      1,015,324 |
| 3.6            |  34.77% |        788,557 |
| 3.5            |  12.99% |        294,623 |
| 3.4            |   6.06% |        137,371 |
| 3.7            |   0.45% |         10,286 |
| 3.3            |   0.36% |          8,128 |
| 2.6            |   0.28% |          6,277 |
| None           |   0.24% |          5,509 |
| 1.17           |   0.06% |          1,449 |
| 3.2            |   0.03% |            580 |

3.3 and 2.6 make up 0.66% of all downloads over the last 90 days

Reply:
@sigmavirus24 Can you clarify if you'd like me to address removal of 2.6 & 3.3 universally or just this one line? I think if the projects official stance is 2.6 & 3.3 are not supported, workarounds and documentation should be removed and updated completely to make this clear to the user. I have no personal needs to continue supporting 2.6 & 3.3, so I don't mind doing this additional work. Please let me know.

Reply:
@sigmavirus24 -- good points. Thank you for those.

@jdufresne  -- If you can update this PR to remove all the Python 2.6 and 3.3 references / support, I think that is the direction we want to go. (tox.ini, setup.py, docs/, etc)